packages/webvh/src/anoncreds/services/WebVhAnonCredsRegistry.ts

```diff
@@ -0,0 +1,568 @@
+        return false
+      }
+
+      // Resolve the verification method to get the public key
+      const didsApi = agentContext.dependencyManager.resolve(DidsApi)
+      const didDocument = await didsApi.resolveDidDocument(proof.verificationMethod as string)
+      
+      if (!didDocument) {
+        agentContext.config.logger.error('Could not resolve verification method DID document')
+        return false
+      }
+
+      // Extract the verification method from the DID document
+      let verificationMethod
+      if ((proof.verificationMethod as string).includes('#')) {
+        const fragment = (proof.verificationMethod as string).split('#')[1]
+        verificationMethod = didDocument.verificationMethod?.find(
+          (vm: any) => vm.id.endsWith(`#${fragment}`)
+        )
+      }
+
+      if (!verificationMethod) {
+        agentContext.config.logger.error('Could not find verification method in DID document')
+        return false
+      }
+
+      // Extract public key from verification method
+      let publicKeyBytes: Uint8Array
+      if ('publicKeyMultibase' in verificationMethod && verificationMethod.publicKeyMultibase) {
+        const publicKeyBuffer = decodeMultibase(verificationMethod.publicKeyMultibase)
+        publicKeyBytes = publicKeyBuffer.data
+      } else {
+        agentContext.config.logger.error('Verification method does not contain a supported public key format')
+        return false
+      }
+
+      // For eddsa-jcs-2022, we need to:
+      // 1. Create a document with the content and proof metadata (without proofValue)
+      // 2. Canonicalize it using JCS (JSON Canonicalization Scheme)
+      // 3. Verify the signature using EdDSA
+
+      // Create the document to be verified (content + proof without proofValue)
+      const documentToVerify = {
+        ...content,
+        proof: {
+          type: proof.type,
+          cryptosuite: proof.cryptosuite,
+          verificationMethod: proof.verificationMethod,
+          proofPurpose: proof.proofPurpose || 'assertionMethod',
+          created: proof.created || new Date().toISOString()
+        }
+      }
+
+      // Canonicalize the document using JCS
+      const canonicalizedDocument = canonicalize(documentToVerify)
+      const documentBytes = TypedArrayEncoder.fromString(canonicalizedDocument)
+
+      // Decode the signature from the proofValue (should be multibase encoded)
+      const signatureBuffer = decodeMultibase(proof.proofValue as string)
+
+      const crypto = new DIDWebvhCrypto(agentContext)
+      const isVerified = await crypto.verify(signatureBuffer.data, documentBytes, publicKeyBytes)
+
+      if (!isVerified) {
+        agentContext.config.logger.error('Signature verification failed')
+        return false
+      }
+
+      agentContext.config.logger.debug('DataIntegrityProof verification successful')
+      return true
+
+    } catch (error) {
+      agentContext.config.logger.error('Error during proof validation', {
```

Comment:
For all error messages here it would be useful to be more specific. If you see this in the context of the full credo agent, it can be hard to understand where a message originates from

```suggestion
      agentContext.config.logger.error('Error during did:webvh resource proof validation', {
```